numpy/tests/test_numpy_version.py

```diff
@@ -0,0 +1,23 @@
+from __future__ import division, absolute_import, print_function
+
+import re
+
+import numpy as np
+from numpy.testing import assert_, run_module_suite
+
+
+def test_valid_numpy_version():
+    # Verify that the numpy version is a valid one (no .post suffix or other
+    # nonsense).  See gh-6431 for an issue caused by an invalid version.
+    version_pattern = "^[0-9]+\.[0-9]+\.[0-9]+(|a[0-9]|b[0-9]|rc[0-9])"
+    dev_suffix = "+(\.dev0\+([0-9a-f]{7}|Unknown))"
```

Comment:
Thanks. ``+`` is incorrect and dev suffix should indeed not be optional, and checking for it without ``?`` makes sure we don't push commits after a release without setting ``ISRELEASED`` back to False (this has happened regularly).

> and accept ``rc1`` etc tags?

That's not wrong I think. The commits between ``rc1`` and ``rc2`` can sensibly be named ``x.y.x.rc2.dev0+abcdef9``. At least that's what we have done regularly, and it's explicitly allowed by PEP 440.